From 9587fb2c57131dd55b285cb92d257fd3819719f7 Mon Sep 17 00:00:00 2001
From: Stefan Seyfried <seife@tuxbox-git.slipkontur.de>
Date: Sun, 9 Sep 2012 16:08:32 +0200
Subject: [PATCH 4/4] aotom: improve scrolling text code

instead of completely scrolling in and out text that's longer than the
display, add a short pause before and after scrolling
---
 frontcontroller/aotom/aotom_main.c |   28 +++++++++++++++++++++++-----
 1 files changed, 23 insertions(+), 5 deletions(-)

diff --git a/frontcontroller/aotom/aotom_main.c b/frontcontroller/aotom/aotom_main.c
index 7859ecf..9fbeb58 100644
--- a/frontcontroller/aotom/aotom_main.c
+++ b/frontcontroller/aotom/aotom_main.c
@@ -211,11 +211,11 @@ static void VFD_clr(void)
 static int draw_thread(void *arg)
 {
 	struct vfd_ioctl_data *data = (struct vfd_ioctl_data *) arg;
-	char buf[sizeof(data->data) + 2 * DISPLAYWIDTH_MAX];
+	char buf[sizeof(data->data) + 0 * 2 * DISPLAYWIDTH_MAX];
 	int len = data->length;
 	int off = 0;
 
-	if (len > YWPANEL_width) {
+	if (0 && len > YWPANEL_width) {
  		memset(buf, ' ', sizeof(buf));
 		off = YWPANEL_width - 1;
 		memcpy(buf + off, data->data, len);
@@ -230,8 +230,17 @@ static int draw_thread(void *arg)
 
 	if(len > YWPANEL_width) {
 		int pos;
-		for(pos = 0; pos < len; pos++) {
-			int i;
+		int i;
+		YWPANEL_VFD_ShowString(buf);
+		// sleep ~500 ms before scrolling
+		for (i = 0; i < 12; i++) {
+			if(kthread_should_stop()) {
+				draw_thread_stop = 1;
+				return 0;
+			}
+			msleep(40);
+		}
+		for(pos = 1; pos <= len - YWPANEL_width; pos++) {
 			if(kthread_should_stop()) {
 				draw_thread_stop = 1;
 				return 0;
@@ -246,11 +255,20 @@ static int draw_thread(void *arg)
 				msleep(40);
 			}
 		}
+		// sleep ~500 ms after scrolling
+		for (i = 0; i < 12; i++) {
+			if(kthread_should_stop()) {
+				draw_thread_stop = 1;
+				return 0;
+			}
+			msleep(40);
+		}
 	}
 
-	clear_display();
 	if(len > 0)
 		YWPANEL_VFD_ShowString(buf + off);
+	else
+		clear_display();
 
 	draw_thread_stop = 1;
 	return 0;
-- 
1.7.7

